scipy/stats/_rvs_sampling.py

```diff
@@ -171,451 +169,3 @@ def rvs_ratio_uniforms(pdf, umax, vmin, vmax, size=1, c=0, random_state=None):
         i += 1
 
     return np.reshape(x, size1d)
-
-
-class NumericalInverseHermite:
-    r"""
-    A Hermite spline fast numerical inverse of a probability distribution.
-
-    The initializer of `NumericalInverseHermite` accepts `dist`, an object
-    representing a continuous distribution, and provides an object with methods
-    that approximate `dist.ppf` and `dist.rvs`. For most distributions,
-    these methods are faster than those of `dist` itself.
-
-    Parameters
-    ----------
-    dist : object
-        Object representing the distribution for which a fast numerical inverse
-        is desired; for instance, a frozen instance of a `scipy.stats`
-        continuous distribution. See Notes and Examples for details.
-    tol : float, optional
-        u-error tolerance (see Notes). The default is 1e-12.
-    max_intervals : int, optional
-        Maximum number of intervals in the cubic Hermite spline used to
-        approximate the percent point function. The default is 100000.
-
-    Attributes
-    ----------
-    intervals : int
-        The number of intervals of the interpolant.
-    midpoint_error : float
```

Comment:
how do we proceed with the deprecation of `midpoint_error`?

Reply:
I will set it to `nan` with a deprecation warning on access as you suggested in https://github.com/scipy/scipy/pull/14703#issuecomment-921990877.

Reply:
Done and a test also added.

Reply:
Have you examined [the NumPy deprecation policy?](https://numpy.org/neps/nep-0023-backwards-compatibility.html) 

> removal of deprecated functionality ... shall be done after at least 2 releases assuming the current 6-monthly release cycle; if that changes, there shall be at least 1 year between deprecation and removal.

Similarly, the [SciPy deprecation policy](https://docs.scipy.org/doc/scipy/reference/dev/core-dev/index.html#deprecations) says that for deprivations there is a two release cycle before removal after proposing the change on the mailing list.

Reply:
Thanks, that works. Let's remove the reference to 1.9.0 then. I suggest to mention the new method `u_error` in the warning message.

Reply:
You are right, @swallan. All the deprecations need 1 year of deprecation cycle. I will change all the warning messages accordingly.

Reply:
@chrisb83 I have added a reference to `u_error` in the deprecation warning of `midpoint_error`.